File: src/main/java/im/tox/tox4j/core/callbacks/FriendNameCallback.java
package im.tox.tox4j.core.callbacks;

import im.tox.tox4j.annotations.NotNull;

public interface FriendNameCallback {

  FriendNameCallback IGNORE = new FriendNameCallback() {

    @Override
    public void friendName(int friendNumber, @NotNull byte[] name) {
    }

  };

  void friendName(int friendNumber, @NotNull byte[] name);

}

File: projects/tox4j/src/main/java/im/tox/tox4j/core/callbacks/FriendNameCallback.scala
package im.tox.tox4j.core.callbacks

import im.tox.tox4j.annotations.NotNull

/**
 * This event is triggered when a friend changes their name.
 */
trait FriendNameCallback {
  /**
   * @param friendNumber The friend number of the friend whose name changed.
   * @param name The new nickname.
   */
  def friendName(friendNumber: Int, @NotNull name: Array[Byte]): Unit = ()
}
